src/anm/analytics.js

```diff
@@ -0,0 +1,77 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics');
+
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
+            } else {
+                var trackUrl = animatronUrl + '?player=' + encodeURIComponent(data);
+                sendViaGif(trackUrl);
+            }
+        } else {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (trackUrl) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = trackUrl;
+        beacon.onerror = beacon.onload = function () {
+            beacon.onerror = beacon.onload = null;
+            setTimeout(event, timeout);
+        }
+    };
+    event();
```

Comment:
It could send data with large intervals (_not_ timeouts) in the background while page is running, if you want to be "safe". And send the remains on `window.unload`. But this will affect page performance. I also don't know why you think `window.unload` could not be called. If it's a crash, I think we don't care, 0,01% of statistics will be lost, I think it's ok (@skavish is it?). It's more important for statistics to be reliable.